### DISCUSSION TITLE
Enable Go to Definition for `@use`/`@forward` and import namespace references

### DISCUSSION THREAD

**Discussion by niksy** [wkillerud/some-sass#253] [Ideas]
Currently only variables (?) have proper "Go to Definition" implementation.

Maybe it would be great to add it for `@use`, `@forward` and import namespace references. This would also align it with how TypeScript works.

Consider following example:

![Screenshot 2024-10-05 at 12 11 56](https://github.com/user-attachments/assets/d418135d-4faf-4571-bf56-fc4e349ce3dd)

Only `$red-color` has "Go to Definition" support. What I would expect is that `variables` and `variables2` also have that (positioned at line 1, column 1 in respective file), and if there is no `as` aliasing, clicking on `@use "variables"` would also take you to that file (currently this is implemented as classic link).

**Comment by wkillerud** [comment]
Functions and mixins also have Go to Definition support (except built-in methods like `math.atan`). But you're right that the namespace does not. I like the idea that it opens the file on line 1  👍 

> If there is no `as` aliasing, clicking on `@use "variables"` would also take you to that file (currently this is implemented as classic link).

I'm not 100% sure what you mean here. Do you mean you want to use Go to Definition on `@use "variables";` and end up in `variables.scss`?

Navigation for `@use`, `@forward` and `@import` are handled by the [document link feature](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_documentLink). In VSCode the interaction is the same (Ctrl/Cmd + click), but the LSP request is technically different. In Sublime the LSP: Follow links is the command to use there.

**Comment by niksy** [comment]
> Navigation for `@use`, `@forward` and `@import` are handled by the [document link feature](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_documentLink). In VSCode the interaction is the same (Ctrl/Cmd + click), but the LSP request is technically different. In Sublime the LSP: Follow links is the command to use there.

Yes, this is it, I’ve mapped special keybinding to at-rule scope so it handles document links same as symbol definition.
